src/app/icd/IcdMonitoringTable.h

```diff
@@ -29,25 +31,54 @@ inline constexpr size_t kIcdMonitoringBufferSize = 40;
 
 struct IcdMonitoringEntry : public PersistentData<kIcdMonitoringBufferSize>
 {
-    static constexpr size_t kKeyMaxSize = 16;
 
     IcdMonitoringEntry(FabricIndex fabric = kUndefinedFabricIndex, NodeId nodeId = kUndefinedNodeId)
     {
         this->fabricIndex      = fabric;
         this->checkInNodeID    = nodeId;
         this->monitoredSubject = nodeId;
     }
+
+    IcdMonitoringEntry(Crypto::SessionKeystore * keyStore, FabricIndex fabric = kUndefinedFabricIndex,
+                       NodeId nodeId = kUndefinedNodeId)
+    {
+        this->fabricIndex      = fabric;
+        this->checkInNodeID    = nodeId;
+        this->monitoredSubject = nodeId;
+        this->sessionKeyStore  = keyStore;
+    }
+
     bool IsValid() { return this->checkInNodeID != kUndefinedNodeId && this->fabricIndex != kUndefinedFabricIndex; }
     CHIP_ERROR UpdateKey(StorageKeyName & key) override;
     CHIP_ERROR Serialize(TLV::TLVWriter & writer) const override;
     CHIP_ERROR Deserialize(TLV::TLVReader & reader) override;
     void Clear() override;
+    CHIP_ERROR SetKey(chip::ByteSpan keyData);
+    CHIP_ERROR DeleteKey(void);
 
-    chip::FabricIndex fabricIndex = static_cast<chip::FabricIndex>(0);
-    chip::NodeId checkInNodeID    = static_cast<chip::NodeId>(0);
-    uint64_t monitoredSubject     = static_cast<uint64_t>(0);
-    chip::ByteSpan key;
-    uint16_t index = 0;
+    /**
+     * @brief Implement the key verification needed by the ICDManagement Server.
+     *        Since we cannot retrieve the key from the AES128KeyHandle IF using
+     *        PSA CRYPTO, we must implement a way to deduct if the verification key
+     *        received is the same or at least works as the same way as the one stored.
+     *
+     *        This method will produce a random number and then encrypt it with the keyData.
+     *        It will then decrypt it with the key stored in the entry. If the resulting decrypted
+     *        challenges matches the randomly generated number, then we can safely assume that both key are interchangeable.
+     *        This method cannot guarantee a perfect match since the probability of two keys generating the same output in AES128 is
+     * not 0 but 1/2^128 which is small enough for our usage.
+     *
+     * @param keyData
+     * @return CHIP_ERROR
```

Comment:
This is not useful documentation.  

Does success mean compatible?  And any error means not?  But then why is this not just returning boolean?

As written, this should be something like EnsureKeyCompatible.